### PR TITLE
fix: use minDisabled, maxDisabled for parameter validation

### LINKED ISSUE #7582: Rich Parameter validation is skipped if min or max is 0.
If you set `validation.min` or `validation.max` to 0 in a rich parameter, we skip validation.

This makes it impossible to correctly enforce a parameter range where 0 is the min or max, e.g. 0-9.

Spotted during a code read:

https://github.com/coder/coder/blob/e951778147c28289cfae32b4a38bc873c98cbbfa/codersdk/richparameters.go#L112

Unfortunately this needs to be resolved at the protobuf / data persistence layer, since we only store the min and max, and not whether either or both were explicitly set in the terraform.

cc @mtojek

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/coder/issues/7582

This PR adjusts the parameter validation to be aligned with recent changes in the [terraform-provider-coder](https://github.com/coder/terraform-provider-coder/releases/tag/v0.8.2) - `MinDisabled` and `MaxDisabled` fields.

### CODE REVIEW

File: provisioner/terraform/testdata/rich-parameters-validation/rich-parameters-validation.tf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    min = 3
+  }
+}
+
+data "coder_parameter" "number_example_max" {
+  name    = "number_example_max"
+  type    = "number"
+  default = 4
+  validation {
+    max = 6
```
Comment (spikecurtis) [inline provisioner/terraform/testdata/rich-parameters-validation/rich-parameters-validation.tf:34]: I'd like to see test cases where the min is explicitly 0 and where the max is explicitly 0.

File: provisioner/terraform/resources.go [hunk 2]
```diff
@@ -483,8 +483,21 @@ func ConvertState(modules []*tfjson.StateModule, rawGraph string, rawParameterNa
...
-			protoParam.ValidationMax = ptrInt32(param.Validation[0].Max)
-			protoParam.ValidationMin = ptrInt32(param.Validation[0].Min)
+
+			// Backward compatibility with terraform-coder-plugin < v0.8.2
+			if _, ok := resource.AttributeValues["min_disabled"]; !ok && param.Validation[0].Min == 0 {
+				param.Validation[0].MinDisabled = true
+			}
+			if _, ok := resource.AttributeValues["max_disabled"]; !ok && param.Validation[0].Max == 0 {
+				param.Validation[0].MaxDisabled = true
+			}
```
Comment (spikecurtis) [inline provisioner/terraform/resources.go:493]: nice.

File: provisioner/terraform/testdata/rich-parameters/rich-parameters.tf [hunk 3]
```diff
@@ -27,6 +27,23 @@ data "coder_parameter" "example" {
...
 }
 
+// Plugin revision v0.7.0 requires defining "min" or "max" rules together.
+data "coder_parameter" "number_example_min_max" {
+  name    = "number_example_min_max"
+  type    = "number"
+  default = 4
+  validation {
+    min = 3
+    max = 6
```
Comment (spikecurtis) [inline provisioner/terraform/testdata/rich-parameters/rich-parameters.tf:37]: can we have a test case where min or max is explicitly 0?

**Review by spikecurtis — APPROVED** [review]
Couple suggestions for interesting test cases, but LGTM.

File: provisioner/terraform/testdata/rich-parameters-validation/rich-parameters-validation.tf (hunk 1 above)
Comment (mtojek) [reply provisioner/terraform/testdata/rich-parameters-validation/rich-parameters-validation.tf:43]: Added

File: provisioner/terraform/testdata/rich-parameters/rich-parameters.tf (hunk 3 above)
Comment (mtojek) [reply provisioner/terraform/testdata/rich-parameters/rich-parameters.tf:37]: Added, although I had to refactor the logic a bit. "min" and "max" need to be specified together as it's a hard requirement by the `coder_parameter` schema

**Pull request merged**